simulation.hpp

```diff
@@ -372,7 +379,6 @@ class Simulation
             for (int j = 0; j < cols_; j++) {
                 if (dispersers(i, j) > 0) {
                     for (int k = 0; k < dispersers(i, j); k++) {
-
                         std::tie(row, col) = dispersal_kernel(generator_, i, j);
```

Comment:
This is where I was thinking of making the change but I think it would be a new type of dispersal kernel that would be called (there might be a simpler way to do this). The new dispersal kernel would need to assign either all dispers at once or remember where it assigned the previous dispers.

Reply:
Given what is in this PR now and our unclear idea about the new dispersal kernel, the kernel should be a different PR from this one. The kernel can be and practically is still the same object in our code, so indeed it can keep some state information through the different calls. We may need to video call to brainstorm this.

Reply:
I agree that a video call would be good for this.